fdbclient/MonitorLeader.actor.cpp

```diff
@@ -161,27 +205,52 @@ std::string trim( std::string const& connectionString ) {
 	return trimmed;
 }
 
-ClusterConnectionString::ClusterConnectionString( std::string const& connectionString ) {
-	auto trimmed = trim(connectionString);
+void ClusterConnectionString::resetToUnresolved() {
+	if (hosts.size() > 0) {
+		coord.clear();
+		hosts.clear();
+		_resolveResults.clear();
+		hasUnresolvedHostnames = false;
```

Comment:
Shouldn't we set hasUnresolvedHostnames to true?

Reply:
Yeah you are right.